packages/scm/src/browser/scm-widget.tsx

```diff
@@ -454,8 +457,8 @@ export class ScmResourceComponent extends ScmElement<ScmResourceComponent.Props>
             onMouseEnter={this.showHover}
             onMouseLeave={this.hideHover}
             ref={this.detectHover}
-            onClick={this.selectChange}
-            onDoubleClick={this.open}>
+            onClick={() => this.handleClick(corePreferences)}
```

Comment:
new function instances is performance killer in react, see https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#no-bind-fn-in-event-handlers

you can access props in the handler function as well

Reply:
Thank you! I've updated the code, I wasn't sure if props could be accessed in the handler functions but now I know :)